### PR TITLE
Tribal crafting expansion

### PR DESCRIPTION
**Pull request by Flakkyboo**
## About The Pull Request
Adds some much needed love to the Tribal crafting tree by adding some new storage options and some new weapons and equipment to make the give people more options later in the round.
overhauls the entire bone sprite set

Adds two new types of bows. 
Adds four new arrow types.  
Adds a new sword and shield.
Adds a new bola.
Tweaked the blowgun.
Adds a new storage option in the arrow quiver.
new sprites for existing bone items (armor, axe spear)
Adds code to make an existing tribal belt to be craftable.
cleans up some code with cotton as currently cloth is not considered a subtype of cotton.
refactor of crafting as introduced in https://github.com/tgstation/tgstation/pull/48451 and the subsequent fix #48477

Sprites generously provided by @SomeAngryMiner and @thatguythere03 

Pulls inspiration and some code from the following
https://github.com/Citadel-Station-13/Citadel-Station-13/pull/11098
https://github.com/Citadel-Station-13/Citadel-Station-13/pull/10993
https://github.com/Citadel-Station-13/Citadel-Station-13/pull/11201
https://github.com/Citadel-Station-13/Citadel-Station-13/pull/11401
https://github.com/Citadel-Station-13/Citadel-Station-13/pull/11516
https://github.com/Citadel-Station-13/Citadel-Station-13/pull/11172
https://github.com/Citadel-Station-13/Citadel-Station-13/pull/11367

## Why It's Good For The Game

The tribal crafting tree is in much need of a expansion its typically only used by ashwalkers and the occasional person looking to dry tobacco or brew drinks. 

This will hopefully give the players that use the tribal crafting more options in terms of weaponry and storage with a clear progression of power as round progresses. New fancy sprites to give tribal crafting a new fresh feel.

image
Above: Examples of the new sprites and items

image
Above: How the new items look when worn and some new storage options

image
Above: The bow will work in 3 steps. you must load the arrow, then draw the arrow back, then fire.

## Changelog
:cl:
add: Added a Goliath shield and it takes less damage from fauna.
add: Added a new Watcher bola, slightly better than the standard bola.
add: Added a new one handed bone sword.
add: Added a new belt quiver to carry arrows.
add: Two new craftable bow types "pipe bow" and "bone bow".
add: 4 new arrow types wood, ash, bone and bronze.
add: New export prices for bows, arrows.
add: New bounty for a bow.
add: New variation of bone armor (heavy bone armor)
add: Cotton seeds in askwalker nest for bowcrafting. 
add: QOL cultivator for ashwalkers added to nest.
tweak: blowgun no longer needs a firing pin and can be used by anyone
soundadd: New sounds for drawing the arrow and firing.
add: New weapon and ammo type for bows.
fix: Added a crafting recipe for the primal explorers belt.
add: A new material Silk which can be made from cloth on the loom. (needed in bowcrafting)
imageadd: Added Sprites for the new items and materials.
imageadd: New sprites for existing bone equipment.
refactor: Cloth is now a subtype of cotton and coded in descending order of creation.
crafting recipes are updated to reflect this.
Cotton>cloth>silk
refactor: Crafting refactor 
/:cl:

### CODE REVIEW

File: code/game/objects/items/shields.dm [hunk 1]
```diff
@@ -44,7 +45,10 @@
 				attackforce = 0
 		else if(isliving(hitby))
 			var/mob/living/L = hitby
-			attackforce = (damage * 2)//simplemobs have an advantage here because of how much these blocking mechanics put them at a disadvantage
+			if(hunter == TRUE)
```
Comment (zeskorion) [inline code/game/objects/items/shields.dm]: make this a bitflag, perhaps

File: code/game/objects/items/weaponry.dm [hunk 2]
```diff
@@ -216,6 +216,28 @@ for further reading, please see: https://github.com/tgstation/tgstation/pull/301
...
+	righthand_file = 'icons/mob/inhands/weapons/swords_lefthand.dmi'
+	slot_flags = ITEM_SLOT_BELT | ITEM_SLOT_BACK
+	force = 15
+	throwforce = 10
+	armour_penetration = 15
+	w_class = WEIGHT_CLASS_NORMAL
+	hitsound = 'sound/weapons/bladeslice.ogg'
+	attack_verb = list("attacked", "slashed", "stabbed", "sliced", "torn", "ripped", "diced", "cut")
+	block_power = 15
+	block_level = 1
```
Comment (zeskorion) [inline code/game/objects/items/weaponry.dm]: i dunno...

File: code/game/objects/items/shields.dm (hunk 1 above)
Comment (Flakkyboo) [reply code/game/objects/items/shields.dm]: ill look into how bitflags work sure to see

File: code/game/objects/items/weaponry.dm (hunk 2 above)
Comment (Flakkyboo) [reply code/game/objects/items/weaponry.dm]: too high of a block power?
I was basing it on the block power of the bone spear which was 25 and knocked 10 off that.

**Comment by Flakkyboo** [comment]
going to be adding the bow framework today.
its going to take a little time

File: code/game/objects/items/shields.dm (hunk 1 above)
Comment (Flakkyboo) [reply code/game/objects/items/shields.dm:48]: Ill look into how bitflags work sure and see if i can make it work

File: code/game/objects/items/weaponry.dm (hunk 2 above)
Comment (Flakkyboo) [reply code/game/objects/items/weaponry.dm:234]: is the block power too high?
I based it on the wielded bonespear which is 25 and knocked 10 off that.

File: code/game/objects/items/shields.dm (hunk 1 above)
Comment (Flakkyboo) [reply code/game/objects/items/shields.dm]: Change implemented

File: code/game/objects/items/weaponry.dm (hunk 2 above)
Comment (Flakkyboo) [reply code/game/objects/items/weaponry.dm]: Removed "Block_level = 1"

**Comment by Flakkyboo** [comment]
Bows are now implemented and tested.
waiting on inhand sprites for shield and bola.
need to add cotton seeds to ashwalker nest for bow creation

**Comment by Flakkyboo** [comment]
No idea what's causing the merge conflict

**Comment by Flakkyboo** [comment]
ah i fucked up on the sprites.

**Comment by Flakkyboo** [comment]
That will be my last commit until reviewed

File: code/game/objects/items/weaponry.dm [hunk 3]
```diff
@@ -216,6 +216,27 @@ for further reading, please see: https://github.com/tgstation/tgstation/pull/301
...
+	righthand_file = 'icons/mob/inhands/weapons/swords_lefthand.dmi'
+	slot_flags = ITEM_SLOT_BELT | ITEM_SLOT_BACK
+	force = 15
+	throwforce = 10
+	armour_penetration = 15
+	w_class = WEIGHT_CLASS_NORMAL
+	hitsound = 'sound/weapons/bladeslice.ogg'
+	attack_verb = list("attacked", "slashed", "stabbed", "sliced", "torn", "ripped", "diced", "cut")
+	block_power = 15
+	block_upgrade_walk = 1
```
Comment (zeskorion) [inline code/game/objects/items/weaponry.dm]: being a subtype of claymore, this only needs the block level of 0 specified, as well as block power (which can be a good bit higher now that you removing passive block. Armor, integrity, sharpness, and flags are unecessary

File: code/modules/projectiles/gun.dm [hunk 4]
```diff
@@ -46,7 +46,7 @@
 	righthand_file = 'icons/mob/inhands/weapons/guns_righthand.dmi'
 
 	var/obj/item/firing_pin/pin = /obj/item/firing_pin //standard firing pin for most guns
-
+	var/no_pin_required = FALSE //whether the gun can be fired without a pin
```
Comment (zeskorion) [inline code/modules/projectiles/gun.dm:49]: I'm not sure, but this may break some shit.

File: code/modules/projectiles/guns/ballistic/bow.dm [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+	weapon_weight = WEAPON_HEAVY //need both hands to fire
+	force = 5
+	mag_type = /obj/item/ammo_box/magazine/internal/bow
+	fire_sound = 'sound/weapons/bowfire.ogg'
+	slot_flags = ITEM_SLOT_BACK
+	item_flags = NONE
+	casing_ejector = FALSE
+	internal_magazine = TRUE
+	pin = null
+	no_pin_required = TRUE
```
Comment (zeskorion) [inline code/modules/projectiles/guns/ballistic/bow.dm:16]: this may break some shit

File: code/modules/projectiles/guns/ballistic/bow.dm [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+	icon_state = "ashenbow"
+	item_state = "ashenbow"
+	force = 8
+
+/obj/item/gun/ballistic/bow/pipe
+	name = "Pipe bow"
+	desc = "A crude projectile weapon made from silk string, pipe and lots of bending."
+	icon_state = "pipebow"
+	item_state = "pipebow"
+	force = 0
```
Comment (zeskorion) [inline code/modules/projectiles/guns/ballistic/bow.dm]: set this to the force of a pipe

File: code/modules/projectiles/guns/ballistic/bow.dm [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+	magazine.get_round(0)
+	update_icon()
+
+/obj/item/gun/ballistic/bow/attack_self(mob/living/user)
+	if (chambered)
+		var/obj/item/ammo_casing/AC = magazine.get_round(0)
+		user.put_in_hands(AC)
+		chambered = null
+		to_chat(user, "<span class='notice'>You gently release the bowstring, removing the arrow.</span>")
+	else if (get_ammo())
```
Comment (zeskorion) [inline code/modules/projectiles/guns/ballistic/bow.dm:36]: maybe a do_mob to pull back the string, which goes back to taught if not in an active hand

File: code/modules/projectiles/projectile/reusable/arrow.dm [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+/obj/item/projectile/bullet/reusable/arrow/ash
+	name = "Ashen arrow"
+	desc = "Fire Hardened arrow."
+	damage = 25
+	ammo_type = /obj/item/ammo_casing/caseless/arrow/ash
+
+/obj/item/projectile/bullet/reusable/arrow/bone //AP for ashwalkers
+	name = "Bone arrow"
+	desc = "An arrow made from bone and sinew."
+	damage = 35
```
Comment (zeskorion) [inline code/modules/projectiles/projectile/reusable/arrow.dm]: I think this is a bit excessive. should be the same as the ashen arrow at most if you are gonna stack that AP on. Miners don't have as much Bullet armor as they have Melee

File: code/modules/projectiles/projectile/reusable/arrow.dm [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+	armour_penetration = 40
+	ammo_type = /obj/item/ammo_casing/caseless/arrow/bone
+
+/obj/item/projectile/bullet/reusable/arrow/bronze
+	name = "Bronze arrow"
+	desc = "Bronze tipped arrow"
+	armour_penetration = 10
```
Comment (zeskorion) [inline code/modules/projectiles/projectile/reusable/arrow.dm:25]: why do bronze arrows suck so much

**Review by zeskorion — CHANGES_REQUESTED** [review]

File: code/modules/projectiles/projectile/reusable/arrow.dm (hunk 8 above)
Comment (Flakkyboo) [reply code/modules/projectiles/projectile/reusable/arrow.dm]: yeah agreed after some tests on different armored targets.
ill make some tweaks for balance

**Comment by Flakkyboo** [comment]
Okay now for the hard stuff.
 1. testing "no_pin_required" variable to make sure it doesn't break guns. (tested. no guns that are accessible and not the bow seem to work)

 2. fixing pin removal recipe not being true.

File: code/modules/projectiles/guns/ballistic/bow.dm (hunk 6 above)
Comment (Flakkyboo) [reply code/modules/projectiles/guns/ballistic/bow.dm]: done

File: code/modules/projectiles/guns/ballistic/bow.dm (hunk 5 above)
Comment (Flakkyboo) [reply code/modules/projectiles/guns/ballistic/bow.dm:16]: Ive tested every gun by firing them and then removing their pin and trying to fire again.
no gun that was not a debug blastcannon fails to fire without a pin.
hopefully it doesn't break other stuff but for now it seems like were okay.

File: code/game/objects/items/weaponry.dm (hunk 3 above)
Comment (Flakkyboo) [reply code/game/objects/items/weaponry.dm]: Done

File: code/modules/projectiles/projectile/reusable/arrow.dm (hunk 8 above)
Comment (Flakkyboo) [reply code/modules/projectiles/projectile/reusable/arrow.dm]: tweaked

File: code/modules/projectiles/projectile/reusable/arrow.dm (hunk 9 above)
Comment (Flakkyboo) [reply code/modules/projectiles/projectile/reusable/arrow.dm:25]: they are a bit better now

File: code/modules/projectiles/guns/ballistic/bow.dm (hunk 7 above)
Comment (Flakkyboo) [reply code/modules/projectiles/guns/ballistic/bow.dm:36]: working on this

**Comment by Flakkyboo** [comment]
Merge conflict caused by clothing repath is resolved.
waiting for review before i commit more

**Comment by Flakkyboo** [comment]
diff bot broke

**Review by zeskorion — DISMISSED** [review]
Even though the bow does not stay strung when not in active hand, making it more like a crossbow, i'll let that slide

**Comment by Flakkyboo** [comment]
Waiting on the final shield inhands before I open this.
New bone armor, bone axe, bone spear sprites incoming.
Need to look into making bone armor similar in function to ash drake armor.
travis is being unfair

**Comment by Flakkyboo** [comment]
need back sprites 
working on conflicts

**Comment by Flakkyboo** [comment]
conflict resolved now all thats left is the back sprites.
i will leave open but do not merge as it

File: code/modules/projectiles/guns/ballistic/bow.dm [hunk 10]
```diff
@@ -0,0 +1,67 @@
...
+	force = 5
+	mag_type = /obj/item/ammo_box/magazine/internal/bow
+	fire_sound = 'sound/weapons/bowfire.ogg'
+	slot_flags = ITEM_SLOT_BACK
+	item_flags = NONE
+	casing_ejector = FALSE
+	internal_magazine = TRUE
+	pin = null
+	no_pin_required = TRUE
+	trigger_guard = TRIGGER_GUARD_NONE //so ashwalkers can use it
```
Comment (Fikou) [inline code/modules/projectiles/guns/ballistic/bow.dm]:
```suggestion
	trigger_guard = TRIGGER_GUARD_ALLOW_ALL //so ashwalkers can use it
```

File: code/modules/projectiles/guns/ballistic/bow.dm (hunk 5 above)
Comment (Fikou) [reply code/modules/projectiles/guns/ballistic/bow.dm:16]: you should make the bamboo syringe gun thing not require a pin

File: code/modules/projectiles/guns/ballistic/bow.dm (hunk 10 above)
Comment (Flakkyboo) [reply code/modules/projectiles/guns/ballistic/bow.dm]: that can be done thank you for the suggestion

**Comment by Flakkyboo** [comment]
ready to be put to vote.
may work on backsprites for bows if i get the time in the week

**Comment by Flakkyboo** [comment]
That's odd I can't see this comment.
I agree that it should and it probably should have a trigger guard compatible with everything

Get Outlook for Android<https://aka.ms/ghei36>
________________________________

**Comment by Flakkyboo** [comment]
okay im done now no more commits im happy where it is.
azlan hated the bow sprite now theres a new one

**Comment by Flakkyboo** [comment]
not sure what caused the bot to fail

**Review by zeskorion — APPROVED** [review]
we'll see how this leaves ashwalkers, but seems fine at the moment.

**Pull request merged**